firebaseMsgSwGenerator.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright 2015-2016, Google, Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const fs = require('fs');
+const template = require('lodash.template');
```

Comment:
Handlebars uses braces (`{{` and `}}`), as a delimiter for it's templates, and it may conflict with JS braces. It seems it's not possible to change delimiters in handlebars (http://stackoverflow.com/questions/14324850/how-to-change-the-default-delimiter-of-handlebars-js). Also, `lodash.template` is already being indirectly included in the project by `sw-precache`, as it uses the same templating engine to build the service-worker.